src/Signal/lid-mapping.ts

```diff
@@ -117,15 +154,21 @@ export class LIDMappingStore {
 		const decoded = jidDecode(lid)
 		if (!decoded) return null
 
-		// Look up reverse user mapping
+		// Check cache first for LID → PN mapping
 		const lidUser = decoded.user
-		// TODO: remove this style and instead load all mappings somehow, and then assign them in the map
-		const stored = await this.keys.get('lid-mapping', [`${lidUser}_reverse`])
-		const pnUser = stored[`${lidUser}_reverse`]
+		let pnUser = this.mappingCache.get(`lid:${lidUser}`)
 
 		if (!pnUser || typeof pnUser !== 'string') {
-			logger.trace(`No reverse mapping found for LID user: ${lidUser}`)
-			return null
+			// Cache miss - check database
+			const stored = await this.keys.get('lid-mapping', [`${lidUser}_reverse`])
+			pnUser = stored[`${lidUser}_reverse`]
```

Comment:
as I discussed before, how about we move the mapping to the new one, but still hit the old one for a few versions?

If the old one is existent, we remove the old key and migrate to the new type.

Reply:
for now we can merge without it